src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -248,6 +225,48 @@ void ExecuteWithAbi (string [] supportedAbis, string apkInputPath, string apkOut
 						count = 0;
 					}
 				}
+
+				HashSet<ulong> deletedEntries = new HashSet<ulong> ();
+				if (apkInputPathExists) {
+					using (var packaged = new ZipArchiveEx (apkInputPath, FileMode.Open)) {
+						foreach (var entry in packaged.Archive) {
+							Log.LogDebugMessage ($"Deregistering item {entry.FullName}");
+							existingEntries.Remove (entry.FullName);
+							if (lastWriteInput <= lastWriteOutput)
+								continue;
+
+							long entryIndexInOutput = -1;
+							if (apk.Archive.ContainsEntry (entry.FullName, out entryIndexInOutput)) {
+								ZipEntry e = apk.Archive.ReadEntry (entry.FullName);
+								// check the CRC values as the ModifiedDate is always 01/01/1980 in the aapt generated file.
+								if (entry.CRC == e.CRC) {
+									Log.LogDebugMessage ($"Skipping {entry.FullName} from {apkInputPath} as its up to date.");
+									continue;
+								}
+							}
+
+							var ms = new MemoryStream ();
+							entry.Extract (ms);
+
+							if (entryIndexInOutput != -1) {
+								Log.LogDebugMessage ($"Refreshing {entry.FullName} from {apkInputPath}");
+								
+								// Force the modified resource to move to the end of the file, by deleting it first so that AddStream adds it
+								// back with a new index. Keeping modified resources toward the end optimizes the delta install for the typical
+								// dev scenario where the user is editing a few resources but most of the APK contents (e.g. the native libs)
+								// don't change and we want to keep their byte offset in the APK fixed. Delta install need not update APK
+								// contents that don't change and don't move.
+								apk.Archive.DeleteEntry ((ulong) entryIndexInOutput);
+								deletedEntries.Add ((ulong) entryIndexInOutput);
+							}
+							else Log.LogDebugMessage ($"Adding {entry.FullName} from {apkInputPath}");
```

Comment:
We normally use braces `{ }` on the `else` if the `if` has them:
```suggestion
							} else {
								Log.LogDebugMessage ($"Adding {entry.FullName} from {apkInputPath}");
							}
```

Reply:
fixed